app/lib/backend_api_logic/service_extension.rb

```diff
@@ -9,6 +9,12 @@ module ServiceExtension
       has_many :backend_apis, through: :backend_api_configs
 
       alias_method :backend_api, :first_backend_api
+
+      has_many :backend_api_metrics, through: :backend_api_configs
+
+      has_many :all_metrics, ->(object) do
```

Comment:
Alternative:

```ruby
def all_metrics
  sql = "((#{metrics.to_sql}) UNION ALL (#{backend_api_metrics.to_sql})) AS metrics"
  Metric.from(sql)
end
```

Reply:
This alternative may not be so good though due to https://www.percona.com/blog/2010/10/25/mysql-limitations-part-3-subqueries/